### PR TITLE
InsightIDR new actions

### PR DESCRIPTION
**Pull request by r7-kszczepanskagorna**
## Proposed Changes
InsightIDR new actions

### Description

Describe the proposed changes:

  -

### Testing

Any testing information goes here.

## PR Requirements

Developers, verify you have completed the following items by checking them off:

### Style

Review the [style guide](https://docs.rapid7.com/insightconnect/style-guide/)

- [ ] For dependencies, pin [OS package](https://docs.rapid7.com/insightconnect/style-guide/#dockerfile) and [Python package](https://docs.rapid7.com/insightconnect/style-guide/#requirements.txt) versions
- [ ] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [ ] For size, use the [slim SDK images](https://docs.rapid7.com/insightconnect/sdk-guide/#sdk-guide) when possible: ``komand/python-3-37-slim-plugin`` and ``komand/python-3-37-plugin``
- [ ] For error handling, use of [PluginException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations/#plugin-exceptions) and [ConnectionTestException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations#connection-exceptions)
- [ ] For logging, use [self.logger](https://docs.rapid7.com/insightconnect/sdk-guide/#logging)
- [ ] For docs, use [changelog style](https://docs.rapid7.com/insightconnect/style-guide/#changelog)
- [ ] For docs, validate markdown with ``make validate`` which calls ``mdl`` to lint ``help.md``

### Functional Checklist
- [ ] Work fully completed
- [ ] Functional
  - [ ] Any new actions/triggers include JSON [test files](https://docs.rapid7.com/insightconnect/style-guide/#tests) in the `tests/` directory created with `icon-plugin run -c sample $action > tests/$action.json`
  - [ ] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [ ] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [ ] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -T all --debug --jq` (use PR format at end)
  - [ ] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -R all --debug --jq` (use PR format at end)

### Run

<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nrapid7/Rapid7 InsightIDR:1.4.0. Step name: get_a_log\n",
    "meta": {},
    "output": {
      "log": {
        "log": {
          "id": "0b9a242d-d2fb-4e42-8656-eb5ff64d652f",
          "name": "Windows Defender",
          "tokens": [
            "bc38a911-65f1-4755-cca3-a330a6336b3a"
          ],
          "structures": [
            "1238a911-65f1-4755-cca3-a330a6336b3a"
          ],
          "user_data": {
            "platform_managed": "true"
          },
          "source_type": "token",
          "token_seed": null,
          "retention_period": "default",
          "links": [
            {
              "rel": "Related",
              "href": "https://example.com"
            }
          ],
          "rrn": "rrn:logsearch:us:bc38a911-65f1-4755-cca3-a330a6336b3a:log:bc38a911-65f1-4755-cca3-a330a6336b3a",
          "logsets_info": [
            {
              "id": "bc38a911-65f1-4755-cca3-a330a6336b3a",
              "name": "Unparsed Data",
              "rrn": "rrn:logsearch:us:bc38a911-65f1-4755-cca3-a330a6336b3a:logset:bc38a911-65f1-4755-cca3-a330a6336b3a",
              "links": [
                {
                  "rel": "Self",
                  "href": "https://example.com/3e966a63-bf3a-4a3c-8903-979c7e90ce85"
                }
              ]
            }
          ]
        }
      }
    }
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/rapid7_insightidr:1.4.0 --debug run < tests/get_a_log.json
</summary>
</details>

<details>

```
{
  "body": {
    "error": "An error occurred during plugin execution!\n\nInsightIDR returned a status code of 404: Not Found ",
    "log": "Connect: Connecting...\nrapid7/Rapid7 InsightIDR:1.4.0. Step name: get_a_log\nNot Found (404): HTTP 404 Not Found\nAn error occurred during plugin execution!\n\nInsightIDR returned a status code of 404: Not Found \nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.7/site-packages/komand-1.0.1-py3.7.egg/komand/plugin.py\", line 311, in handle_step\n    output = self.start_step(input_message['body'], 'action', logger, log_stream, is_test, is_debug)\n  File \"/usr/local/lib/python3.7/site-packages/komand-1.0.1-py3.7.egg/komand/plugin.py\", line 419, in start_step\n    output = func(params)\n  File \"/usr/local/lib/python3.7/site-packages/rapid7_insightidr_rapid7_plugin-1.4.0-py3.7.egg/komand_rapid7_insightidr/actions/get_a_log/action.py\", line 24, in run\n    'get')\n  File \"/usr/local/lib/python3.7/site-packages/rapid7_insightidr_rapid7_plugin-1.4.0-py3.7.egg/komand_rapid7_insightidr/util/resource_helper.py\", line 69, in resource_request\n    raise PluginException(f'InsightIDR returned a status code of {response.status_code}: {status_code_message}')\nkomand.exceptions.PluginException: An error occurred during plugin execution!\n\nInsightIDR returned a status code of 404: Not Found \n",
    "meta": {},
    "status": "error"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/rapid7_insightidr:1.4.0 --debug run < tests/get_a_log_bad.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nrapid7/Rapid7 InsightIDR:1.4.0. Step name: get_all_logs\n",
    "meta": {},
    "output": {
      "logs": {
        "logs": [
          {
            "id": "0b9a242d-d2fb-4e42-8656-eb5ff64d652f",
            "links": [
              {
                "href": "https://example/log_search/management/logs/20d5a090-04d0-4807-ac21-fdd49926232f/topkeys",
                "rel": "Related"
              }
            ],
            "logsets_info": [
              {
                "id": "20d5a090-04d0-4807-ac21-fdd49926232f",
                "links": [
                  {
                    "href": "https://example/log_search/management/logsets/20d5a090-04d0-4807-ac21-fdd49926232f",
                    "rel": "Self"
                  }
                ],
                "name": "Asset Authentication",
                "rrn": "rrn:logsearch:us:20d5a090-04d0-4807-ac21-fdd49926232f:logset:20d5a090-04d0-4807-ac21-fdd49926232f"
              }
            ],
            "name": "Endpoint Agents",
            "retention_period": "default",
            "rrn": "rrn:logsearch:us:20d5a090-04d0-4807-ac21-fdd49926232f:log:20d5a090-04d0-4807-ac21-fdd49926232f",
            "source_type": "token",
            "structures": [
              "20d5a090-04d0-4807-ac21-fdd49926232f"
            ],
            "token_seed": null,
            "tokens": [
              "20d5a090-04d0-4807-ac21-fdd49926232f"
            ],
            "user_data": {
              "platform_managed": "true"
            }
          },
          {
            "id": "20d5a090-04d0-4807-ac21-fdd49926232f",
            "links": [
              {
                "href": "https://us.rest.logs.insight.rapid7.com/log_search/management/logs/a2aba094-94ef-4487-932f-57e1a7d36d87/topkeys",
                "rel": "Related"
              }
            ],
            "logsets_info": [
              {
                "id": "20d5a090-04d0-4807-ac21-fdd49926232f",
                "links": [
                  {
                    "href": "https://exampl.com/log_search/management/logsets/20d5a090-04d0-4807-ac21-fdd49926232f",
                    "rel": "Self"
                  }
                ],
                "name": "Internal Logs",
                "rrn": "rrn:logsearch:us:20d5a090-04d0-4807-ac21-fdd49926232f:logset:20d5a090-04d0-4807-ac21-fdd49926232f"
              }
            ],
            "name": "Log Updates",
            "retention_period": "default",
            "rrn": "rrn:logsearch:us:20d5a090-04d0-4807-ac21-fdd49926232f:log:20d5a090-04d0-4807-ac21-fdd49926232f",
            "source_type": "internal",
            "structures": [
              "20d5a090-04d0-4807-ac21-fdd49926232f"
            ],
            "token_seed": null,
            "tokens": [],
            "user_data": {}
          },
          {
            "id": "0b9a242d-d2fb-4e42-8656-eb5ff64d652f",
            "links": [
              {
                "href": "https://example/log_search/management/logs/20d5a090-04d0-4807-ac21-fdd49926232f/topkeys",
                "rel": "Related"
              }
            ],
            "logsets_info": [
              {
                "id": "20d5a090-04d0-4807-ac21-fdd49926232f",
                "links": [
                  {
                    "href": "https:/example/log_search/management/logsets/20d5a090-04d0-4807-ac21-fdd49926232f",
                    "rel": "Self"
                  }
                ],
                "name": "Internal Logs",
                "rrn": "rrn:logsearch:us:20d5a090-04d0-4807-ac21-fdd49926232f:logset:20d5a090-04d0-4807-ac21-fdd49926232f"
              }
            ],
            "name": "Alert Audit Log",
            "retention_period": "default",
            "rrn": "rrn:logsearch:us:20d5a090-04d0-4807-ac21-fdd49926232f:log:20d5a090-04d0-4807-ac21-fdd49926232f",
            "source_type": "internal",
            "structures": [
              "fa6a4440-4579-4a03-be08-c259a84db062"
            ],
            "token_seed": null,
            "tokens": [],
            "user_data": {}
          }
            }
          }
        ]
      }
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/rapid7_insightidr:1.4.0 --debug run < tests/get_all_logs.json
</summary>
</details>

<details>

```
[*] Use ``make menu`` for available targets
[*] Including available Makefiles: ../tools/Makefiles/Helpers.mk ../tools/Makefiles/Colors.mk
--
[*] Running validators
[*] Validating plugin at .

[*] Running Integration Validators...
[*] Executing validator HelpValidator
[*] Executing validator ChangelogValidator
[*] Executing validator RequiredKeysValidator
[*] Executing validator UseCaseValidator
[*] Executing validator SpecPropertiesValidator
[*] Executing validator SpecVersionValidator
[*] Executing validator FilesValidator
[*] Executing validator TagValidator
[*] Executing validator DescriptionValidator
[*] Executing validator TitleValidator
[*] Executing validator VendorValidator
[*] Executing validator DefaultValueValidator
[*] Executing validator IconValidator
[*] Executing validator RequiredValidator
[*] Executing validator VersionValidator
[*] Executing validator DockerfileParentValidator
[*] Executing validator LoggingValidator
[*] Executing validator ProfanityValidator
[*] Executing validator AcronymValidator
[*] Executing validator JSONValidator
[*] Executing validator OutputValidator
[*] Executing validator RegenerationValidator
[*] Executing validator HelpInputOutputValidator
[*] Executing validator SupportValidator
[*] Executing validator RuntimeValidator
[*] Executing validator VersionPinValidator
[*] Executing validator ExampleInputValidator
[*] Plugin failed validation! The following validation errors occurred:

Validator "ExampleInputValidator" failed!
        Cause: All inputs should have example field in plugin.spec.
In action "list_investigations": input "index", there is no example field.


----
[*] Total time elapsed: 1278.41ms

[*] Validating spec with js-yaml
[SUCCESS] Passes js-yaml spec check


[*] Validating markdown...
[SUCCESS] Passes markdown linting

[*] Validating python files for style...
./komand_rapid7_insightidr/actions/assign_user_to_investigation/action.py:8:1: E302 expected 2 blank lines, found 1
./komand_rapid7_insightidr/actions/assign_user_to_investigation/action.py:29:9: F841 local variable 'e' is assigned to but never used
./unit_test/test_assign_user_to_investigation.py:5:1: E402 module level import not at top of file
./unit_test/test_assign_user_to_investigation.py:6:1: E402 module level import not at top of file
./unit_test/test_assign_user_to_investigation.py:7:1: E402 module level import not at top of file
./unit_test/test_assign_user_to_investigation.py:8:1: E402 module level import not at top of file
./unit_test/test_assign_user_to_investigation.py:9:1: E402 module level import not at top of file
./unit_test/test_assign_user_to_investigation.py:26:9: F841 local variable 'e' is assigned to but never used
./unit_test/test_assign_user_to_investigation.py:29:1: W293 blank line contains whitespace
./unit_test/test_assign_user_to_investigation.py:30:110: W291 trailing whitespace
./unit_test/test_assign_user_to_investigation.py:36:9: E303 too many blank lines (2)
[FAIL] Fails flake8 linting

[*] Validating python files for security vulnerabilities...
[main]  INFO    profile include tests: None
[main]  INFO    profile exclude tests: None
[main]  INFO    cli include tests: None
[main]  INFO    cli exclude tests: None
[main]  INFO    running on Python 3.7.3
Run started:2021-01-04 17:51:54.800989

Test results:
        No issues identified.

Code scanned:
        Total lines of code: 2456
        Total lines skipped (#nosec): 0

Run metrics:
        Total issues (by severity):
                Undefined: 0.0
                Low: 0.0
                Medium: 0.0
                High: 0.0
        Total issues (by confidence):
                Undefined: 0.0
                Low: 0.0
                Medium: 0.0
                High: 0.0
Files skipped (0):
[SUCCESS] Passes bandit security checks

[*] Checking for typos with Misspell...
[SUCCESS] Passes Misspell check
```

<summary>
make validate
</summary>
</details>

<details>

```
[*] Validating plugin with all validators at .

[*] Running Integration Validators...
[*] Executing validator HelpValidator
[*] Executing validator ChangelogValidator
[*] Executing validator RequiredKeysValidator
[*] Executing validator UseCaseValidator
[*] Executing validator SpecPropertiesValidator
[*] Executing validator SpecVersionValidator
[*] Executing validator FilesValidator
[*] Executing validator TagValidator
[*] Executing validator DescriptionValidator
[*] Executing validator TitleValidator
[*] Executing validator VendorValidator
[*] Executing validator DefaultValueValidator
[*] Executing validator IconValidator
[*] Executing validator RequiredValidator
[*] Executing validator VersionValidator
[*] Executing validator DockerfileParentValidator
[*] Executing validator LoggingValidator
[*] Executing validator ProfanityValidator
[*] Executing validator AcronymValidator
[*] Executing validator JSONValidator
[*] Executing validator OutputValidator
[*] Executing validator RegenerationValidator
[*] Executing validator HelpInputOutputValidator
[*] Executing validator SupportValidator
[*] Executing validator RuntimeValidator
[*] Executing validator VersionPinValidator
[*] Executing validator ExampleInputValidator
[*] Executing validator ExceptionValidator
[*] Executing validator CredentialsValidator
[*] Executing validator PasswordValidator
[*] Executing validator PrintValidator
[*] Executing validator ConfidentialValidator
[*] Executing validator DockerValidator
[*] Executing validator URLValidator
WARNING: URLs found that return a 4xx code. Verify they are publicly accessible and if not, update with a working URL.
violation: plugin.spec.yaml[366]: https://us.api.insight.rapid7.com
violation: plugin.spec.yaml[367]: https://us.api.insight.rapid7.com
violation: help.md[88]: https://example.com/3e966a63-bf3a-4a3c-8903-979c7e90ce85
violation: help.md[157]: https://example.com/3e966a63-bf3a-4a3c-8903-979c7e90ce85
[*] Plugin failed validation! The following validation errors occurred:

Validator "ExampleInputValidator" failed!
        Cause: All inputs should have example field in plugin.spec.
In action "list_investigations": input "index", there is no example field.


----
[*] Total time elapsed: 38122.369999999995ms
```

<summary>
icon-validate --all .
</summary>
</details>

### CODE REVIEW

File: rapid7_insightidr/help.md [hunk 1]
```diff
@@ -36,6 +36,134 @@ Example input:
 
 ### Actions
 
+#### Get All Logs
+
+This action is used to request used to list all Logs for an account.
```
Comment (jschipp-r7) [inline rapid7_insightidr/help.md:41]:
```suggestion
This action is used to request a list of all Logs for an account.
```

File: rapid7_insightidr/help.md [hunk 2]
```diff
@@ -36,6 +36,134 @@ Example input:
...
+        }
+      ]
+    }
+  }
+}
+```
+
+#### Get a Log
+
+This action is used to request used to get a specific Log from an account.
```
Comment (jschipp-r7) [inline rapid7_insightidr/help.md:100]:
```suggestion
This action is used to get a specific log from an account.
```

File: rapid7_insightidr/plugin.spec.yaml [hunk 3]
```diff
@@ -494,3 +537,28 @@ actions:
         description: Events from logs
         type: "[]events"
         required: true
+  get_a_log:
+    title: Get a Log
+    description: Request used to get a specific Log from an account
```
Comment (jschipp-r7) [inline rapid7_insightidr/plugin.spec.yaml:542]:
```suggestion
    description: Get a specific log from an account
```

File: rapid7_insightidr/plugin.spec.yaml [hunk 4]
```diff
@@ -494,3 +537,28 @@ actions:
...
+      id:
+        title: ID
+        description: Query ID
+        type: string
+        example: 174e4f99-2ac7-4481-9301-4d24c34baf06
+        required: true
+    output:
+      log:
+        title: Log
+        description: Specific log from an account
```
Comment (jschipp-r7) [inline rapid7_insightidr/plugin.spec.yaml:553]:
```suggestion
        description: Requested log
```

File: rapid7_insightidr/plugin.spec.yaml [hunk 5]
```diff
@@ -494,3 +537,28 @@ actions:
...
+        type: logsets_info
+        required: true
+  get_all_logs:
+    title: Get All Logs
+    description: Request used to list all Logs for an account
```
Comment (jschipp-r7) [inline rapid7_insightidr/plugin.spec.yaml:558]:
```suggestion
    description: Request used to list all logs for an account
```

File: rapid7_insightidr/plugin.spec.yaml [hunk 6]
```diff
@@ -494,3 +537,28 @@ actions:
...
+    output:
+      logs:
+        title: Logs
+        description: Specific logs from an account
```
Comment (jschipp-r7) [inline rapid7_insightidr/plugin.spec.yaml:562]:
```suggestion
        description: All logs
```

**Review by jschipp-r7 — APPROVED** [review]
